dhall/tests/Dhall/Test/Dhall.hs

```diff
@@ -239,3 +246,64 @@ shouldHandleUnionsCorrectly =
         resolvedExpression <- Dhall.Import.assertNoImports parsedExpression
 
         Dhall.Core.denote resolvedExpression @=? Dhall.embed Dhall.inject value
+
+shouldConvertDhallToHaskellCorrectly :: TestTree
+shouldConvertDhallToHaskellCorrectly =
+    testGroup
+        "Marshall Dhall code to Haskell"
+        [ "True" `correspondsTo` True
+        , "False" `correspondsTo` False
+        , "2" `correspondsTo` (2 :: Natural)
+        , "+2" `correspondsTo` (2 :: Integer)
+        , "2.0" `correspondsTo` (2.0 :: Double)
+        , "2.0" `correspondsTo` (2.0 :: Scientific)
+        , "\"ABC\"" `correspondsTo` ("ABC" :: Data.Text.Text)
+        , "\"ABC\"" `correspondsTo` ("ABC" :: Data.Text.Lazy.Text)
+        , "\"ABC\"" `correspondsTo` ("ABC" :: String)
+        , "Some 2" `correspondsTo` (Just 2 :: Maybe Natural)
+        , "None Natural" `correspondsTo` (Nothing :: Maybe Natural)
+        , "[ 2, 3, 5 ]" `correspondsTo` ([ 2, 3, 5 ] :: Seq Natural)
+        , "[ 2, 3, 5 ]" `correspondsTo` ([ 2, 3, 5 ] :: [Natural])
+        , "[ 2, 3, 5 ]" `correspondsTo` ([ 2, 3, 5 ] :: Vector Natural)
+        , "[] : List Natural" `correspondsTo` ([] :: [Natural])
+        , "{=}" `correspondsTo` ()
+        , "{ _1 = True, _2 = {=} }" `correspondsTo` (True, ())
+        ]
+  where
+    correspondsTo :: (Eq a, Interpret a, Show a) => Text -> a -> TestTree
+    dhallCode `correspondsTo` expectedHaskellValue =
+      testCase "Marshall Dhall code to Haskell" $ do
+          actualHaskellValue <- Dhall.input Dhall.auto dhallCode
+
+          expectedHaskellValue @=? actualHaskellValue
+
+shouldConvertHaskellToDhallCorrectly :: TestTree
+shouldConvertHaskellToDhallCorrectly =
+    testGroup
+        "Marshall Haskell to Dhall code"
+        [ "True" `correspondsTo` True
+        , "False" `correspondsTo` False
+        , "2" `correspondsTo` (2 :: Natural)
+        , "+2" `correspondsTo` (2 :: Integer)
+        , "2.0" `correspondsTo` (2.0 :: Double)
+--      , "2.0" `correspondsTo` (2.0 :: Scientific)
```

Comment:
Why is this commented out?

Reply:
Missing instance, which I can add for symmetry

Reply:
Can you add a comment regarding the missing instance?